### PR TITLE
Added defaultPlaceholder

### LINKED ISSUE #149: Enable custom placeholder in `KBarSearch`
When there is no action being performed, the default placeholder is ```"Type a command or search…"```, but we'd like to use a custom placeholder

### PR DESCRIPTION
**Pull request by BigOsvaap**
Added a function parameter that allows adding a custom place holder.

Fix #149

### CODE REVIEW

**Comment by BigOsvaap** [comment]
My bad u.u, I think I solved. Example of displays now:

### Default placeholder
Screen Shot 2022-01-07 at 14 25 34

### Custom placeholder
Screen Shot 2022-01-07 at 14 26 40

Let me know what you think, I'll be pending. 

Regards!

**Comment by BigOsvaap** [comment]
Thinking a little more about another solution, I just could rename de const `placeholder` in both places in the original file.

https://github.com/timc1/kbar/blob/d8049e9b58ba62e8c9febbd26bb32e63bc1ba515/src/KBarSearch.tsx#L35

```Javascript
const defaultPlaceholder = React.useMemo( 
```

https://github.com/timc1/kbar/blob/d8049e9b58ba62e8c9febbd26bb32e63bc1ba515/src/KBarSearch.tsx#L54

```Javascript
 placeholder={defaultPlaceholder} 
```

and just modify 

https://github.com/timc1/kbar/blob/d8049e9b58ba62e8c9febbd26bb32e63bc1ba515/src/KBarSearch.tsx#L35-L41

with this

```Javascript
const defaultPlaceholder = React.useMemo(
    (): string => {
      const text = props.placeholder ?? "Type a command or search…";
      return currentRootActionId
        ? actions[currentRootActionId].name
        : text;
    }, [actions, currentRootActionId]);
```

and ends with the same result as above but the users won't be able to do something as you did in the https://github.com/timc1/kbar/issues/149#issuecomment-1005755308 I don't if that is expected, with the current implementation the users would be able to modify the placeholder.

File: src/KBarSearch.tsx [hunk 1]
```diff
@@ -5,8 +5,12 @@ import { useKBar } from "./useKBar";
 export const KBAR_LISTBOX = "kbar-listbox";
 export const getListboxItemId = (id: number) => `kbar-listbox-item-${id}`;
 
+interface CustomAttributes<T> extends React.InputHTMLAttributes<T> {
+  defaultPlaceholder?: string | undefined 
+}
+
 export function KBarSearch(
-  props: React.InputHTMLAttributes<HTMLInputElement>
+  props: CustomAttributes<HTMLInputElement>,
```
Comment (timc1) [inline src/KBarSearch.tsx:13]: Nit: can we just inline the type here:

```suggestion
  props: React.InputHTMLAttributes<HTMLInputElement> & { defaultPlaceholder?: string }
```

File: src/KBarSearch.tsx [hunk 2]
```diff
@@ -33,12 +37,12 @@ export function KBarSearch(
   }, [currentRootActionId, query]);
 
   const placeholder = React.useMemo(
-    () =>
-      currentRootActionId
+    (): string => {
+      const text =  props.defaultPlaceholder ?? "Type a command or search…";
```
Comment (timc1) [inline src/KBarSearch.tsx:41]:
```suggestion
      const default =  props.defaultPlaceholder ?? "Type a command or search…";
```

File: src/KBarSearch.tsx [hunk 3]
```diff
@@ -33,12 +37,12 @@ export function KBarSearch(
...
+      return currentRootActionId
         ? actions[currentRootActionId].name
-        : "Type a command or search…",
-    [actions, currentRootActionId]
-  );
+        : text;
```
Comment (timc1) [inline src/KBarSearch.tsx:44]:
```suggestion
        : default;
```

File: src/KBarSearch.tsx [hunk 4]
```diff
@@ -33,12 +37,12 @@ export function KBarSearch(
...
+    }, [actions, currentRootActionId]);
```
Comment (timc1) [inline src/KBarSearch.tsx:45]:
```suggestion
    }, [actions, currentRootActionId, props.defaultPlaceholder]);
```

**Review by timc1 — COMMENTED** [review]
Almost there 😉

File: src/KBarSearch.tsx (hunk 3 above)
Comment (BigOsvaap) [reply src/KBarSearch.tsx]: I'm not able to use default as the name of the value because it's a keyword, but I applied all the other changes.

**Comment by BigOsvaap** [comment]
I made the changes, but I wasn't able to change the value name from text to default because it's a keyword. Thanks for the review 👍

File: src/KBarSearch.tsx [hunk 5]
```diff
@@ -33,12 +33,12 @@ export function KBarSearch(
   }, [currentRootActionId, query]);
 
   const placeholder = React.useMemo(
-    () =>
-      currentRootActionId
+    (): string => {
+      const defaultText =  props.defaultPlaceholder ?? "Type a command or search…";
```
Comment (timc1) [inline src/KBarSearch.tsx:37]:
```suggestion
      const defaultText = props.defaultPlaceholder ?? "Type a command or search…";
```

**Review by timc1 — APPROVED** [review]

**Pull request merged**